src/api/app/helpers/webui/comments_helper.rb

```diff
@@ -27,21 +31,40 @@ def roles_for_package(comment)
 
   def roles_for_request(comment)
     roles = []
-    roles << 'Submitter' if comment.commentable.creator == comment.user.login
-    comment.commentable.bs_request_actions.inject(roles) do |acc, action|
-      acc += Relationship
-             .joins(:role)
-             .joins(:project)
-             .where(projects: { name: action.target_project })
-             .where(user: comment.user)
-             .pluck('roles.title')
-
-      acc + Relationship
-            .joins(:role)
-            .joins(:package)
-            .where(packages: { name: action.target_package })
-            .where(user: comment.user)
-            .pluck('roles.title')
+    roles << 'author' if comment.commentable.creator == comment.user.login
+    roles << 'reviewer' if comment.commentable.reviews.pluck(:user_id).include?(comment.user.id) ||
+                           comment.commentable.reviews.pluck(:group_id).intersect?(comment.user.groups.ids)
+    source_roles = []
+    target_roles = []
+    comment.commentable.bs_request_actions.each do |action|
+      source_roles += Relationship.joins(:role).joins(:project)
+                                  .where(projects: { name: action.source_project })
+                                  .where(user: comment.user)
+                                  .where('roles.title': 'maintainer')
+                                  .pluck('roles.title')
+                                  .union(Relationship.joins(:role).joins(:package).joins('LEFT JOIN projects ON projects.id = packages.project_id')
+                                                     .where(packages: { name: action.source_package })
+                                                     .where(projects: { name: action.source_project })
+                                                     .where(user: comment.user)
+                                                     .where('roles.title': 'maintainer')
+                                                     .pluck('roles.title'))
+
+      target_roles += Relationship.joins(:role).joins(:project)
+                                  .where(projects: { name: action.target_project })
+                                  .where(user: comment.user)
+                                  .where('roles.title': 'maintainer')
+                                  .pluck('roles.title')
+                                  .union(Relationship.joins(:role).joins(:package).joins('LEFT JOIN projects ON projects.id = packages.project_id')
+                                                     .where(packages: { name: action.target_package })
+                                                     .where(projects: { name: action.target_project })
+                                                     .where(user: comment.user)
+                                                     .where('roles.title': 'maintainer')
+                                                     .pluck('roles.title'))
```

Comment:
All of this should go into `queries/relationships_finder.rb`

Reply:
done